pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -155,7 +155,20 @@ func (a *index) podWorkloadBuilder(
 		if (!IsPodRunning(p) && !IsPodPending(p)) || p.Spec.HostNetwork {
 			return nil
 		}
-		podIP, err := netip.ParseAddr(p.Status.PodIP)
+		k8sPodIPs := p.Status.PodIPs
+		if len(k8sPodIPs) == 0 && p.Status.PodIP != "" {
```

Comment:
question, when Status.PodIPs can be empty?  I guess we donot need to check it since have checked Running status before

Reply:
This is mostly defense in depth against weird Pods, shouldn't really happen. In particular our tests often just set podIp and not PodIps. 

But any Pod can be arbitrarily modified by user, so we cannot assume it's valid

Reply:
Also IIRC the k8s docs say that _if_ PodIPs is non-empty, the first entry in the list is PodIP - they don't declare any guarantees that PodIPs _will_ be non-empty. I think this is mostly for back compat with old K8S versions though.